src/pepr/operator/controllers/network/policies.ts

```diff
@@ -111,6 +129,23 @@ export async function networkPolicies(pkg: UDSPackage, namespace: string, istioM
     // Generate the policy
     const keycloakGeneratedPolicy = generate(namespace, keycloakPolicy);
     policies.push(keycloakGeneratedPolicy);
+
+    // Ambient mode: add waypoint-to-istiod egress policy for the package's dedicated waypoint
+    if (istioMode === IstioState.Ambient && shouldUseAmbientWaypoint(pkg)) {
```

Comment:
nit: isn't checkin istioMode here redudant since `shouldUseAmbientWaypoint(pkg)` does that too?
```suggestion
    if (shouldUseAmbientWaypoint(pkg)) {
```

Reply:
missed this one too!